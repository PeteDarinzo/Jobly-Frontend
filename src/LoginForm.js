import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import { Button, Form, FormGroup, Label, Input} from 'reactstrap';
import "./Form.css"

const LoginForm = ({ login }) => {

  const initialState = {
    username: "",
    password: "",
  }

  const [formData, setFormData] = useState(initialState);
  const history = useHistory();

  const handleChange = (e) => {
    e.preventDefault();
    const { name, value } = e.target;
    setFormData(formData => ({ ...formData, [name]: value }));
  }

  const handleSubmit = (e) => {
    e.preventDefault();
    login(formData);
    setFormData(initialState);
    history.push("/");
  }


  return (
      <Form onSubmit={handleSubmit} className="Form shadow">
        <h1>Log In</h1>
        <FormGroup>
          <Label for="username">Username</Label>
          <Input
            type="text"
            name="username"
            id="username"
            placeholder="username"
            onChange={handleChange}
          />
        </FormGroup>
        <FormGroup>
          <Label for="password">Password</Label>
          <Input
            type="password"
            name="password"
            id="examplePassword"
            placeholder="password"
            onChange={handleChange}
          />
        </FormGroup>
        <Button color="primary" className="btn-lg">Submit</Button>
      </Form>
  );
}

export default LoginForm;