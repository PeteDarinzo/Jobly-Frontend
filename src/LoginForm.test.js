import { render } from '@testing-library/react';
import { MemoryRouter } from "react-router-dom"
import LoginForm from "./LoginForm";

it("renders signup form without crashing", () => {
  render(
    <MemoryRouter>
      <LoginForm login={() => { }} error="" />
    </MemoryRouter>
  );
});

it("matches the home snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <LoginForm login={() => { }} error="" />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});