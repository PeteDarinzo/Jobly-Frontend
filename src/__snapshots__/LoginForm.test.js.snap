// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the home snapshot 1`] = `
<DocumentFragment>
  <div>
    <form
      class="Form shadow"
    >
      <h1>
        Log In
      </h1>
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="username"
        >
          Username
        </label>
        <input
          class="form-control"
          id="username"
          name="username"
          placeholder="username"
          type="text"
        />
      </div>
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="password"
        >
          Password
        </label>
        <input
          class="form-control"
          id="password"
          name="password"
          placeholder="password"
          type="password"
        />
      </div>
      <button
        class="btn-lg btn btn-primary"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
